stdlib/collections/__init__.pyi

```diff
@@ -283,22 +282,23 @@ class Counter(dict[_T, int], Generic[_T]):
         def __eq__(self, other: object) -> bool: ...
         def __ne__(self, other: object) -> bool: ...
 
-    def __add__(self, other: Counter[_T]) -> Counter[_T]: ...
-    def __sub__(self, other: Counter[_T]) -> Counter[_T]: ...
-    def __and__(self, other: Counter[_T]) -> Counter[_T]: ...
-    def __or__(self, other: Counter[_T]) -> Counter[_T]: ...  # type: ignore[override]
+    def __add__(self, other: Counter[_S]) -> Counter[_T | _S]: ...
+    def __sub__(self, other: Counter[Any]) -> Counter[_T]: ...
+    def __and__(self, other: Counter[Any]) -> Counter[_T]: ...
```

Comment:
I don't think we should change to `Any`, unless users complain. With `Any`, you won't get an error for `counter_of_ints - counter_of_strings`, which is bad IMO. With `_T`, you will get an error for `counter_of_strings - counter_of_strings_or_nones`, which apparently doesn't come up in practice because nobody has complained about it.

Reply:
The same argument surely applies for `Abstract Set.__sub__`:
https://github.com/python/typeshed/blob/c987c78077dc21ef2b1fdb11b83eeb947c3b4276/stdlib/typing.pyi#L841

Are we saying that in an ideal world, that also wouldn't be `AbstractSet[Any]`, but it would probably be too disruptive to change it now?

Reply:
Changing `AbstractSet` would be more disruptive, but the argument doesn't apply to `AbstractSet` as much as it does to `Counter`s:
- It's common to have sets that contain `None`s and subtract between them, which I think is the most common use for subtracting sets of different types. I use sets containing `None`s all the time.
- A counter that contains `None`s would be something like `{None: 2, "foo": 1, "bar": 6}`, which seems like it could useful in some case but not very often. I don't remember ever using a counter like this, for example.

Reply:
I'm still sorta unconvinced. I'll revert for now, and maybe open an issue to discuss in more depth later :)